File: 2598. Smallest Missing Non-negative Integer After Operations/Solution.java
class Solution {
    public int findSmallestInteger(int[] nums, int value) {
        int[] counter = new int[value];
        for (int n : nums) {
            if (n < 0) {
                if (n % value == 0) {
                    counter[-(n % value)]++;
                } else {
                    counter[value - (-n % value)]++;
                }
            } else {
                counter[n % value]++;
            }
        }
        int low = nums.length, pos = 0;
        for (int i = 0; i < value; i++) {
            if (counter[i] == 0) return i;
            if (counter[i] < low) {
                low = counter[i];
                pos = i;
            }
        }
        return value * low + pos;
    }
}

File: 2598. Smallest Missing Non-negative Integer After Operations/Solution.py
class Solution:
    def findSmallestInteger(self, nums: List[int], value: int) -> int:
        counter = [0 for i in range(value)]
        for n in nums:
            if n < 0:
                if n % value == 0:
                    counter[0] += 1
                else:
                    counter[value - (-n % value)] += 1
            else:
                counter[n % value] += 1
        low, pos = len(nums), 0
        for i in range(value):
            if counter[i] == 0: return i
            if counter[i] < low:
                low, pos = counter[i], i
        return value * low + pos